addons/autoLock/autoLock.lua

```diff
@@ -0,0 +1,166 @@
+-- All rights reserved.
+-- 
+-- Redistribution and use in source and binary forms, with or without
+-- modification, are permitted provided that the following conditions are met:
+-- 
+-- * Redistributions of source code must retain the above copyright
+--   notice, this list of conditions and the following disclaimer.
+-- * Redistributions in binary form must reproduce the above copyright
+--   notice, this list of conditions and the following disclaimer in the
+--   documentation and/or other materials provided with the distribution.
+-- * Neither the name of autoLock nor the names of its contributors may be 
+--   used to endorse or promote products derived from this software without
+--   specific prior written permission.
+
+-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+-- ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+-- WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+-- DISCLAIMED. IN NO EVENT SHALL ReaperX BE LIABLE FOR ANY
+-- DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+-- (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+-- LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+-- ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+-- SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+--
+--
+-- Automatically equips a gear set and uses the /lockstyle command when leaving town.
+-- This addon requires the Gearswap addon.
+--
+-- To use this addon create a gear set titled sets.lockstyle in your job.lua file.
+-- Ex:
+-- 		sets.lockstyle = {
+--			head="Raider's Bonnet +2",
+--			body="Raider's Vest +2",hands="Raider's Armlets +2",
+--			legs="Raider's Culottes +2",feet="Raider's Poulaines +2"}
+--
+--
+
+_addon.name = "autoLock"
+_addon.author = "bangerang"
+_addon.version = "1.0"
+_addon.commands = {'al', 'autolock'}
+_addon.language = 'english'
+
+require('luau')
+config = require('config')
+
+defaults = {}
+defaults.zoneDelay = 10
+defaults.keyBind = '^x'
+defaults.autoLock = true
+defaults.ignoreZones = S{
+    "Ru'Lude Gardens", "Upper Jeuno", "Lower Jeuno", "Port Jeuno", "Port Windurst", "Windurst Waters", "Windurst Woods", "Windurst Walls",
+    "Heavens Tower", "Port San d'Oria", "Northern San d'Oria", "Southern San d'Oria", "Chateau d'Oraguille", "Port Bastok", "Bastok Markets",
+    "Bastok Mines", "Metalworks", "Aht Urhgan Whitegate", "Tavnazian Safehold", "Nashmau", "Selbina", "Mhaura", "Norg", "Rabao", "Kazham",
+    "Eastern Adoulin", "Western Adoulin", "Leafallia", "Celennia Memorial Library", "Mog Garden"}
+
+settings = config.load(defaults)
+
+tokens = {}
+tokens.lockstyle = false
+
+windower.send_command('bind %s al':format(settings.keyBind))
+windower.send_command('input /lockstyle off')
+
+function al_output(msg)
```

Comment:
You may consider using the `logger` library and it's `log` function, which basically does the same thing as this. Only it uses color `207` by default, but users can adjust that in its settings file so it outputs to the color they like.